docs/components.rst

```diff
@@ -124,31 +124,77 @@ Content
 Product
 +++++++
 
-The user has to give a name to the package and URL from where that package is going to be fetched.
-It has two fields **Add New** and **View Existing**.
+While provisioning a virtual machine, packages can be added to a virtual machine. A single repository is identified under the term **Product**.
+
+* Create New
+
+    Consists of two fields
+
+    * Product Name
+        * The repository will be recognized with the product name instead of the repository URL.
+        * Mapping a repository URL to a name, makes identifying a repository URL with the help of product name easy.
+    * Product URL
+        * The location from where the repository for a package can be added.
+    
+    .. note ::
+        A single product name will hold only one URL of a repository not more than that.
+    
+* View Existing
+
+    All existing products are enlisted here along with their repository URL.
 
 ++++
 View
 ++++
 
-View is a layer of abstraction where we can encapsulate two or more products into a single view.
-Products added by user will be visible here and user has to give name to the view and select the product of it's choice from the list.
+A single view consists of multiple products along with their corressponding repository URLs. 
+
+* Create New
+
+    Consists of 
+
+        * View Name 
+            * Mutiple products will be recognized with a single name, **View Name**.
+            * If a view is selected, all the underlying products consisted in that view are added. 
+
+        * Select Products
+            * To create a view, one or more products can be selected.
+            * The view will now consist of the selected products.
+
+* View Existing
+
+    All existing view are enlisted here along with the  included products and their corressponding repository URLs.
```

Comment:
There is extra space between `the` and `included` 
Correct the spelling of corresponding
Use Content View instead of just view.